google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -1080,6 +1093,7 @@ def query query, params: nil, external: nil, max: nil, cache: true,
                       params: params }
           options[:dataset] ||= self
           updater = QueryJob::Updater.from_options service, query, options
+          updater.location = location if location # may be dataset reference
```

Comment:
I know why you are using a conditional, but users may decide that they want to set the value to `nil`, so we should probably have `QueryJob::Updater#location=` do the right thing when setting the value to `nil`.